core/trino-main/src/main/java/io/trino/dispatcher/LocalDispatchQuery.java

```diff
@@ -307,6 +311,14 @@ public boolean isInfoPruned()
         return stateMachine.isQueryInfoPruned();
     }
 
+    @Override
+    public long getRetainedSizeInBytes()
+    {
+        return INSTANCE_SIZE
+                + stateMachine.getRetainedSizeInBytes()
+                + MoreSizeOf.ATOMIC_BOOLEAN_INSTANCE_SIZE; // notificationSentOrGuaranteed
```

Comment:
why you have sizeOf(DateTime) (method instead of a constant), but size of atomic is a constant?

i like the sizeof(AtomicBoolean) signature because it frees us from a code comment and the compiler checks the reference

the downside of sizeof(AtomicBoolean) is when calculating size of collections ( we can do `io.airlift.slice.SizeOf#estimatedSizeOf(java.util.Map<K,V>, long, long)` if we know a thing is constant)

wdyt?